### PR TITLE
Remove obsolete build infrastructure

### PR DESCRIPTION
**Pull request by natemcmaster**
* Remove obsolete targets which are unnecessary now that this repo no longer builds git submodules in a separate build process
* Remove the need for static analysis of 'ArtifactInfo' items
* Simplify how the code signing task is configured
* Remove unused repo tasks
* Remove duplicate lists of external dependencies and packages to be produced
* Remove obsolete build definition
* Remove obsolete build script parameters
~~* Stop producing a NuGet package for some testasset assemblies used for integration testing~~

### CODE REVIEW

File: src/Servers/test/FunctionalTests/ServerComparison.FunctionalTests.csproj [hunk 1]
```diff
@@ -9,7 +9,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <Reference Include="Microsoft.AspNetCore.Server.IntegrationTesting.IIS" />
+    <ProjectReference Include="$(RepositoryRoot)src\Servers\IIS\IntegrationTesting.IIS\src\Microsoft.AspNetCore.Server.IntegrationTesting.IIS.csproj" />
```
Comment (Tratcher) [inline src/Servers/test/FunctionalTests/ServerComparison.FunctionalTests.csproj]: Because it's not a shipping package?

Comment (natemcmaster) [reply src/Servers/test/FunctionalTests/ServerComparison.FunctionalTests.csproj]: Correct. This assembly is a test asset, not something we ship to customers. It was only put into a package as a way to share test code between repos. That's not necessary anymore.

**Comment by natemcmaster** [comment]
:up: :date: Chatted with @pakrym. Apparently the IntegrationTesting and IntegrationTesting.IIS packages are still used in benchmark infrastructure some how. I've revert the changes which would have stopped producing those as packages.

**Review by ryanbrandenburg — APPROVED** [review]
I don't see any problems here.

**Review by dougbu — COMMENTED** [review]
The general simplification of our infrastructure looks good.

Is there any file or area that needs a particularly deep review?

File: src/Mvc/benchmarkapps/BasicApi/BasicApi.csproj [hunk 2]
```diff
@@ -17,6 +18,8 @@
     <PackageReference Include="MySqlConnector" Version="0.43.0" />
     <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="2.1.1.1" />
     <PackageReference Include="Pomelo.EntityFrameworkCore.MySql" Version="2.1.1" />
+    <!-- This suppresses errors about using PackageReference instead of Reference. For unclear reasons, this project has pinned dependencies. -->
```
[the remark this answers is not in the document]
Comment (natemcmaster) [reply src/Mvc/benchmarkapps/BasicApi/BasicApi.csproj]: Ok. I'll update the comment.

**Comment by natemcmaster** [comment]
> Is there any file or area that needs a particularly deep review?

@dougbu - no one particular file. As you can see, this touched many files. This change is primarily about deleting old stuff. The best thing to review is the build outputs to make sure we're getting the same results.

I'm holding off on this change until Preview 2 has stabilized, so likely won't merge till next week.

**Pull request merged**